hal/src/rtl872x/ble_hal.cpp

```diff
@@ -1020,23 +1018,49 @@ void BleEventDispatcher::bleEventDispatchThread(void *context) {
 
 void BleGap::bleCommandThread(void *context) {
     BleGap* gap = (BleGap*)context;
+    uint8_t command = 0;
     while (true) {
-        uint8_t command;
+        bool locked = false;
         if (!os_queue_peek(gap->cmdQueue_, &command, CONCURRENT_WAIT_FOREVER, nullptr)) {
             LOCAL_DEBUG("---> Enter ble cmd thread, cmd: %d", command);
             if (command == BLE_CMD_STOP_ADV || command == BLE_CMD_STOP_ADV_NOTIFY) {
                 if (gap->isAdvertising()) {
                     LOCAL_DEBUG( "stopAdvertising in ble cmd thread");
-                    BleLock lk;
+                    // FIXME: duplication, but for now works as a workaround
+                    while (!(locked = s_bleMutex.lock(BLE_WAIT_CMD_THREAD_TRY_LOCK)) && !gap->bleCommandThreadExit_) {
+                        // Nothing
+                    }
+                    SCOPE_GUARD({
+                        if (locked) {
+                            s_bleMutex.unlock();
+                        }
+                    });
+                    if (gap->bleCommandThreadExit_) {
+                        LOCAL_DEBUG("Exit ble cmd thread");
+                        break;
+                    }
                     gap->stopAdvertising();
                     LOCAL_DEBUG("notifyAdvStop in ble cmd thread");
                     if (command == BLE_CMD_STOP_ADV_NOTIFY) {
                         gap->notifyAdvStop();
                     }
+                    hal_ble_unlock(nullptr);
```

Comment:
This is not necessary since we have `SCOPE_GUARD()` above to unlock.

Reply:
Yup, forgot to remove after refactoring a bit. Fixed.